addons/mail/static/src/gif_picker/gif_picker.js

```diff
@@ -0,0 +1,141 @@
+/* @odoo-module */
+
+import { useOnBottomScrolled } from "@mail/utils/hooks";
+import { Component, onWillStart, useState } from "@odoo/owl";
+import { useService, useAutofocus } from "@web/core/utils/hooks";
+import { removeFromArrayWithPredicate } from "@mail/utils/arrays";
+
+/**
+ * @typedef {Object} Props
+ * @property {import("@mail/core/thread_model").Thread} thread
+ * @property {Function} [close]
+ * @extends {Component<Props, Env>}
+ */
+export class GifPicker extends Component {
+    static template = "mail.GifPicker";
+    static props = ["thread", "close?"];
+
+    async setup() {
+        this.rpc = useService("rpc");
+        /** @type {import('@mail/core/thread_service').ThreadService} */
+        this.threadService = useService("mail.thread");
+        this.userService = useService("user");
+        useAutofocus();
+        useOnBottomScrolled("scroller", () => {
+            this.search(true);
+        });
+        this.state = useState({
+            favorites: [],
+            searchTerm: "",
+            gifs: [],
+            next: "",
+            showCategories: true,
+            categories: [],
+            loadingGif: false,
+            loadingError: false,
+        });
+        onWillStart(this.loadCategories);
+        onWillStart(this.loadFavorite);
+    }
+
+    get evenGif() {
+        return this.state.gifs.filter((gif, index) => index % 2 === 0);
+    }
+
+    get oddGif() {
+        return this.state.gifs.filter((gif, index) => index % 2 !== 0);
+    }
+
+    async loadCategories() {
+        try {
+            const { tags } = await this.rpc("/mail/gif/categories", {
+                locale: this.userService.lang,
+                country: this.userService.lang.slice(3, 5),
+            });
+            if (tags) {
+                this.state.categories = tags;
+            }
+        } catch {
+            this.state.loadingError = true;
+        }
+    }
+
+    openCategories() {
+        this.state.showCategories = true;
+        this.state.searchTerm = "";
+    }
+
+    closeCategories() {
+        this.state.showCategories = false;
+    }
+
+    async search(concat = false) {
```

Comment:
shouldn't it be debounced? maybe wait a few ms without typing before sending the request.
when you type a long search term and clear it by keeping backspace pressed, it's possible to reach too many simultaneous http requests

Reply:
Yes, I forgot about that :)